File: pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    //cash owed
    float x;
    //coins back
    int qa = 0;
    int da = 0;
    int na = 0;
    int pa = 0;
    //remainder
    int r;
    //control variable
    int w;
    //coins
    int q = 25; //quarter
    int d = 10; //dime
    int n = 5; //nickel
    int p = 1; //penny

    do
    {
        //ask how much change is owed
        x = get_float("change owed?: ");
    }
    while (x < 0);

    int c = round(x * 100); //new cash owed in cents!!

    if (c >= q)
    {
        r = c % q;
        w = c - r;
        qa =qa + (w / q);
        c = r;
        printf("%i quarters\n", qa);
    }
    if (c >= d)
    {
        r = c % d;
        w = c - r;
        da = da + (w / d);
        c = r;
        printf("%i dimes\n", da);
    }
    if (c >= n)
    {
        r = c % n;
        w = c - r;
        na = na + (w / n);
        c = r;
        printf("%i nickels\n", na);
    }
    if (c >= p)
    {
        r = c % p;
        w = c - r;
        pa = pa + (w / p);
        c = r;
        printf("%i pennies\n", pa);
    }
}

File: pset6/cash/cash.py
from cs50 import get_float

while True:
    c = get_float("change owed?: ")
    if c > 0:
        break

#transform change into cents
c = c * 100

quarters = 0
dimes = 0
nickels = 0
pennies = 0

if c >= 25:
    rq = c % 25
    w = c - rq
    quarters = quarters + w / 25
    c = rq
    
if c >= 10:
    rd = c % 10
    w = c - rd
    dimes = dimes + w / 10
    c = rd
    
if c >= 5:
    rn = c % 5
    w = c - rn
    nickels = nickels + w / 5
    c = rn
    
if c >= 1:
    rp = c % 1
    w = c - rp
    pennies = pennies + w / 1
    c = rp

print(f"{int(quarters + dimes + nickels + pennies)}")
